.github/workflows/reusable-windows-ci.yml

```diff
@@ -0,0 +1,288 @@
+          colcon_build_args: ${{ inputs.colcon-args }}
+          colcon_build_args_default: --event-handlers console_direct+
+          cmake_args: ${{ inputs.cmake-args }}
+          cmake_args_default: '-T ${{ inputs.vs-toolset }} ${{ env.colcon-build-default-cmake-args }}'
+          cmake_build_type: ${{ matrix.cmake-build-type }}
+          workspace: ${{ github.workspace }}
+
+      - name: Upload build artifacts
+        uses: eProsima/eProsima-CI/external/upload-artifact@v0
+        with:
+          name: fastdds_python_build_${{ inputs.label }}
+          path: ${{ github.workspace }}
+
+
+  fastdds_python_test:
+    needs: fastdds_python_build
+    runs-on: ${{ inputs.os-version }}
+    if: ${{ inputs.run-tests == true }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake-build-type:
+          - 'RelWithDebInfo'
+    steps:
+      - name: Download python build artifacts
+        uses: eProsima/eProsima-CI/external/download-artifact@v0
+        with:
+          name: fastdds_python_build_${{ inputs.label }}
+          path: ${{ github.workspace }}
+
+      - name: Install Fix Python version
+        uses: eProsima/eProsima-CI/external/setup-python@v0
+        with:
+          python-version: '3.11'
+
+      - name: Get minimum supported version of CMake
+        uses: eProsima/eProsima-CI/external/get-cmake@v0
+        with:
+          cmakeVersion: '3.22.6'
+
+      - name: Install OpenSSL
+        uses: eProsima/eprosima-CI/windows/install_openssl@v0
+        with:
+          version: '3.1.1'
+
+      - name: Update OpenSSL environment variables
+        run: |
+          # Update the environment
+          if (Test-Path -Path $Env:ProgramW6432\OpenSSL)
+          {
+            "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+          }
+          elseif (Test-Path -Path $Env:ProgramW6432\OpenSSL-Win)
+          {
+            "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL-Win" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+          }
+          elseif (Test-Path -Path $Env:ProgramW6432\OpenSSL-Win64)
+          {
+            "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL-Win64" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+          }
+          else
+          {
+            Write-Error -Message "Cannot find OpenSSL installation."
+            exit 1
+          }
+
+      - name: Install colcon
+        uses: eProsima/eProsima-CI/windows/install_colcon@v0
+
+      - name: Install Python dependencies
+        uses: eProsima/eProsima-CI/windows/install_python_packages@v0
+        with:
+          packages: vcstool xmlschema pywin32 pytest
+
+      - name: Install swig
+        shell: pwsh
+        run: choco install swig --allow-downgrade --version=4.0.2.04082020
+
+      - name: Install doxygen
+        uses: ssciwr/doxygen-install@v1
+
+      - name: Prepare build meta file
```

Comment:
I think we can have an eprosima-ci action for this and use it everywhere (only supporting .metas as .yaml)